src/backend/utils/html/replace-redundant-breaklines.js

```diff
@@ -0,0 +1,5 @@
+const redundantBreaklineRegex = /(<br>((\s?)+)){2,}|((<p>|<div>)(\s+)?(<br>)(\s+)?(<\/p>|<\/div>)((\s?)+)){1,}/g;
```

Comment:
Regex to deal with the DOM makes me nervous.  Can you not use CSS Selectors for this?  The DOM already has a parsing engine built into it without us inventing a new one via regex.

There's a famous StackOverflow answer about the perils of this: https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags?answertab=active#tab-top